src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -0,0 +1,987 @@
+                                | None -> yield reference
+                        ]
+
+                    tcConfigB.referencedDLLs <- []
+
+                    tcConfigB.primaryAssembly <-
+                        (if loadClosure.UseDesktopFramework then
+                             PrimaryAssembly.Mscorlib
+                         else
+                             PrimaryAssembly.System_Runtime)
+                    // Add one by one to remove duplicates
+                    dllReferences
+                    |> List.iter (fun dllReference -> tcConfigB.AddReferencedAssemblyByPath(dllReference.Range, dllReference.Text))
+
+                    tcConfigB.knownUnresolvedReferences <- loadClosure.UnresolvedReferences
+                | None -> ()
+
+            setupConfigFromLoadClosure ()
+
+            let tcConfig = TcConfig.Create(tcConfigB, validate = true)
+            let _outfile, _, assemblyName = tcConfigB.DecideNames sourceFiles
+
+            // Resolve assemblies and create the framework TcImports. This caches a level of "system" references. No type providers are
+            // included in these references.
+
+            let frameworkDLLs, nonFrameworkResolutions, unresolvedReferences =
+                TcAssemblyResolutions.SplitNonFoundationalResolutions(tcConfig)
+
+            let frameworkDLLsKey =
+                frameworkDLLs
+                |> List.map (fun ar -> ar.resolvedPath) // The cache key. Just the minimal data.
+                |> List.sort // Sort to promote cache hits.
+
+            // Prepare the frameworkTcImportsCache
+            //
+            // The data elements in this key are very important. There should be nothing else in the TcConfig that logically affects
+            // the import of a set of framework DLLs into F# CCUs. That is, the F# CCUs that result from a set of DLLs (including
+            // FSharp.Core.dll and mscorlib.dll) must be logically invariant of all the other compiler configuration parameters.
+            let key =
+                FrameworkImportsCacheKey(
+                    frameworkDLLsKey,
+                    tcConfig.primaryAssembly.Name,
+                    tcConfig.GetTargetFrameworkDirectories(),
+                    tcConfig.fsharpBinariesDir,
+                    tcConfig.langVersion.SpecifiedVersion
+                )
+
+            let! tcGlobals, frameworkTcImports =
+                FrameworkImportsCache.Get(key, ComputeFrameworkImports tcConfig frameworkDLLs nonFrameworkResolutions)
+
+            // Note we are not calling diagnosticsLogger.GetDiagnostics() anywhere for this task.
+            // This is ok because not much can actually go wrong here.
+            let diagnosticsLogger =
+                CompilationDiagnosticLogger("nonFrameworkAssemblyInputs", tcConfig.diagnosticsOptions)
+
+            use _ = new CompilationGlobalsScope(diagnosticsLogger, BuildPhase.Parameter)
+
+            // TODO: might need to put something like this somewhere
```

Comment:
Yes all inputs should be somehow versioned and contribute to project snapshot identity.

Reply:
This is now handled by `ReferencedOnDisk` in project snapshot.